examples/perplexity/perplexity.cpp

```diff
@@ -120,6 +120,83 @@ void perplexity(llama_context * ctx, const gpt_params & params) {
     printf("\n");
 }
 
+void perplexity_lines(llama_context * ctx, const gpt_params & params) {
+    // Calculates perplexity over each line of the prompt
+
+    std::vector<std::string> prompt_lines;
+
+    size_t pos=0;
+    while( pos < params.prompt.size() ) {
+        std::string line;
+        while( true ) {
+            if( params.prompt[pos] == '\n' || pos == params.prompt.size() )
+                break;
+            line += params.prompt[pos++];
+        }
+        pos++;
+        prompt_lines.push_back( line );
+    }
```

Comment:
```suggestion
    size_t pos = 0;
    while (pos < params.prompt.size()) {
        std::string line;
        while (true) {
            if (params.prompt[pos] == '\n' || pos == params.prompt.size()) {
                break;
            }
            line += params.prompt[pos++];
        }
        pos++;
        prompt_lines.push_back(line);
    }
```

Btw, `stringstream` + `getline()` should be better here